docs/pages/radiogroup.js

```diff
@@ -0,0 +1,695 @@
+}: {|
+  generatedDocGen: {| [string]: DocGen |},
+|}): Node {
+  return (
+    <Page title="RadioGroup">
+      <PageHeader
+        name="RadioGroup"
+        description={generatedDocGen.RadioGroup?.description}
+        defaultCode={`
+      function RadioButtonExample() {
+        const [favorite, setFavorite] = React.useState(undefined);
+
+        return (
+          <RadioGroup legend="Gender">
+            <RadioGroup.RadioButton
+              checked={favorite === 'Female'}
+              id="genderFemale"
+              label="Female"
+              name="gender-pref"
+              onChange={() => setFavorite( 'Female' )}
+              value="Female"
+            />
+            <RadioGroup.RadioButton
+              checked={favorite === 'Male'}
+              id="genderMale"
+              label="Male"
+              name="gender-pref"
+              onChange={() => setFavorite( 'Male' )}
+              value="Male"
+            />
+            <RadioGroup.RadioButton
+              checked={favorite === 'Non-binary'}
+              id="genderNon-binary"
+              label="Non-binary"
+              name="gender-pref"
+              onChange={() => setFavorite( 'Non-binary' )}
+              value="Non-binary"
+            />
+            <RadioGroup.RadioButton
+              checked={favorite === 'Prefer not to state'}
+              id="genderPrefer not to state"
+              label="Prefer not to state"
+              name="gender-pref"
+              onChange={() => setFavorite( 'Prefer not to state' )}
+              value="Prefer not to state"
+            />
+          </RadioGroup>
+        );
+      }
+      `}
+      />
+
+      <GeneratedPropTable generatedDocGen={generatedDocGen.RadioGroup} />
+      <GeneratedPropTable
+        Component={RadioGroup?.RadioButton}
+        name="RadioGroup.RadioButton"
+        id="RadioGroup.RadioButton"
+        generatedDocGen={generatedDocGen.RadioGroupButton}
+      />
+
+      <MainSection name="Usage guidelines">
+        <MainSection.Subsection columns={2}>
+          <MainSection.Card
+            cardSize="md"
+            type="do"
+            title="When to use"
+            description={`
+          - In a list, form or a Table, to present users with multiple, related options where only one option can be selected.
+          - When selection doesn’t take immediate effect and requires Form submission.
+        `}
+          />
+          <MainSection.Card
+            cardSize="md"
+            type="don't"
+            title="When not to use"
+            description={`
+          - Situations where users can select multiple options. Use [Checkbox](/checkbox) instead.
+          - When there is only one item to select or deselect. Use Checkbox instead.
+          - When a selection takes immediate effect, especially on mobile. Use [Switch](/switch) instead.
+          - When visually, it’s hard to tell that a RadioGroup turns something on or off. Use Switch instead.
```

Comment:
This is a little awkward. Consider something like "When it is visually difficult to observe that RadioGroup turns something on or off."